### PR TITLE
Formula dataview refactor

### PR DESCRIPTION
**Pull request by dgaubert**
Lots of ES6's syntax sugar goodies applied:
  - Template strings to build formula query.
  - New class syntax
  - Renamed base class
  - Remove every reference to widgets.

### CODE REVIEW

File: test/acceptance/dataviews/overviews.js [hunk 1]
```diff
@@ -88,7 +102,14 @@ describe('dataviews using tables without overviews', function() {
                     if (err) {
                         return done(err);
                     }
-                    assert.deepEqual(formula_result, { operation: 'count', result: 256, nulls: 0, type: 'formula' });
+                    assert.deepEqual(formula_result, {
+                        operation: 'count',
+                        result: 256,
+                        nulls: 0,
+                        infinities: 0,
```
Comment (rochoa) [inline test/acceptance/dataviews/overviews.js]: Why do we need to change the tests this being a refactor?

Comment (dgaubert) [reply test/acceptance/dataviews/overviews.js]: Actually they are bad tests. All of them was missing special float values. As we do in other dataviews. So I fixed them.

Comment (rochoa) [reply test/acceptance/dataviews/overviews.js]: It feels weird to me, as only the expectation is changing. In these cases, I usually prefer to do two PR: one with the tests changed the second one with the refactor.

Comment (dgaubert) [reply test/acceptance/dataviews/overviews.js]: You are right, my mind was in rush when I figured out what was going on here.

File: lib/cartodb/models/dataview/formula.js [hunk 2]
```diff
@@ -1,36 +1,44 @@
...
-    '  {{=it._column}} != \'-infinity\'::float',
-    'AND',
-    '  {{=it._column}} != \'NaN\'::float{{?}}'
-].join('\n'));
-
-var VALID_OPERATIONS = {
+const BaseDataview = require('./base');
+const debug = require('debug')('windshaft:dataview:formula');
+
+const countInfinitiesQueryTpl = ctx => `
```
Comment (oleurud) [inline lib/cartodb/models/dataview/formula.js]: :rocket:

File: lib/cartodb/models/dataview/formula.js [hunk 3]
```diff
@@ -1,36 +1,44 @@
...
+    SELECT count(1) FROM (${ctx.query}) __cdb_formula_infinities
+    WHERE ${ctx.column} = 'infinity'::float OR ${ctx.column} = '-infinity'::float
+`;
+
+const countNansQueryTpl = ctx => `
```
Comment (oleurud) [inline lib/cartodb/models/dataview/formula.js]: :rocket:

**Pull request merged**